workspace/gh-workflow-rs/src/workflow.rs

```diff
@@ -473,9 +476,8 @@ mod tests {
     fn split(content: &str) -> (String, String) {
         let parsed = Workflow::parse(content).unwrap();
         let actual = parsed.to_string().unwrap();
-        let expected =
-            serde_yaml::to_string(&serde_yaml::from_str::<Value>(content).unwrap()).unwrap();
-
+        // Normalize line endings to Unix format
+        let expected = content.replace("\r\n", "\n").trim_end().to_string();
```

Comment:
not required. Tests are passing with the original changes.

Reply:
@tusharmath This is required. I have done logging with their hex codes. So I have introduced this.  Even both expected and actual are same the test fails because Windows uses `\r\n` while Unix-based systems use `\n`. This is causing issue particularly to fail.

I think you have done on MAC OS maybe it passes there but in Windows and Linux it failing I tried on both